src/brick.c

```diff
@@ -40,22 +40,33 @@ static void assert_brick_callback(struct pg_brick *brick)
  *
  * @brick: the struct pg_brick we are initializing
  */
-static void alloc_edges(struct pg_brick *brick)
+static int alloc_edges(struct pg_brick *brick, struct pg_error **errp)
 {
 	enum pg_side i;
+	bool is_other_side_empty = false;
 
 	for (i = 0; i < MAX_SIDE; i++) {
 		struct pg_brick_side *side = &brick->sides[i];
 
 		if (brick->type == PG_MULTIPOLE) {
 			side->nb = 0;
-			g_assert(side->max);
-			side->edges = g_new0(struct pg_brick_edge, side->max);
+			if (side->max == 0) {
+				if (is_other_side_empty) {
+					*errp = pg_error_new(
+						"Brick,two sides empties");
```

Comment:
maybe you can write the name of the brick.
I'll use something like: 
"Brick '%s' need to be able to have at last 1 neighbour"